### ISSUE TITLE
cpp.Stars get de-referenced when passed to Pointer.addressOf in Haxe 4.0.0-preview.5

### ISSUE THREAD

**Issue by Aidan63** [HaxeFoundation/hxcpp#741]
Hello,
I've recently been testing my libraries to ensure they work with preview 5 but noticed that my directx externs no longer work. The reasons seems to be that `cpp.Star` now get de-referenced when passed into `Pointer.addressOf()`.

```
var int = 5;
var ptr    : cpp.Star<Int>           = cast cpp.Pointer.addressOf(int);
var ptrptr : cpp.Star<cpp.Star<Int>> = cast cpp.Pointer.addressOf(ptr);

trace(ptr, ptrptr);
```

In preview 4 the above code compiles fine and `src/Main.hx:17: Pointer(0096FAB8),Pointer(0096FABC)` gets output as expected. But when attempting to compile with preview 5 MSVC fails to compile with the following output.

```
Error: Main.cpp
./src/Main.cpp(18): error C2440: 'type cast': cannot convert from 'cpp::Pointer<int>' to 'int **'
./src/Main.cpp(18): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
```
Looking at the generated code you can see that `ptr` gets de-referenced before being passed to `addressOf`.
```
HXLINE(  12)		int _hx_int = 5;
HXLINE(  14)		int * ptr = ( (int *)(::cpp::Pointer_obj::addressOf(_hx_int)) );
HXLINE(  15)		int * * ptrptr = ( (int * *)(::cpp::Pointer_obj::addressOf((*(ptr)))) );
```
When reverting to preview 4 the following c++ gets generated and ptr isn't de-referenced.
```
HXLINE(  12)		int _hx_int = 5;
HXLINE(  14)		int * ptr = ::cpp::Pointer_obj::addressOf(_hx_int);
HXLINE(  15)		int * * ptrptr = ::cpp::Pointer_obj::addressOf(ptr);
```
Trying to use `cpp.RawPointer<cpp.Star<T>>` to get the pointer to pointer also generates the same error. Not sure if this change was intentional, but if so whats now the best way to get a pointer to pointer for a `cpp.Star<T>`.

Cheers

**Comment by Aidan63** [comment]
I finally got around to looking into what was causing this and found the issue! The issue is in haxe's gencpp.ml not hxcpp.

This line, https://github.com/HaxeFoundation/haxe/commit/f26ead193d0ea5cfc15cbe1ebc1b280a3b62bdc6#diff-7807454c794194a587a88cd72d90d13eR3228 introduced the issue.

I removed the line and compiled haxe myself and the example code compiles as expected. I would submit a PR for this but until an hour ago my ocaml knowledge was zero (and still basically) so I'm not sure of the effects of removing the line or why it was added in the first place, but all unit tests still pass with the change and I've compiled a few of my projects and they all still work.

Cheers
